In [4]:
import pandas as pd

In [13]:
loan = '/content/lending_club_loan_dataset.csv'
ld = pd.read_csv(loan)
ld.head()

id grade  annual_inc  short_emp  emp_length_num home_ownership  \
0  11454641     A    100000.0          1               1           RENT   
1   9604874     A     83000.0          0               4            OWN   
2   9684700     D     78000.0          0              11       MORTGAGE   
3   9695736     D     37536.0          0               6       MORTGAGE   
4   9795013     D     65000.0          0              11       MORTGAGE   

     dti             purpose        term  last_delinq_none  \
0  26.27         credit_card   36 months                 1   
1   5.39         credit_card   36 months                 0   
2  18.45  debt_consolidation   60 months                 1   
3  12.28             medical   60 months                 0   
4  11.26  debt_consolidation   36 months                 0   

   last_major_derog_none  revol_util  total_rec_late_fee  od_ratio  bad_loan  
0                    NaN        43.2                 0.0  0.160624         0  
1                    NaN        21.5                 0.0  0.810777         0  
2                    NaN        46.3                 0.0  0.035147         1  
3                    NaN        10.7                 0.0  0.534887         1  
4                    NaN        15.2                 0.0  0.166500         0

In [14]:
ld.shape

(20000, 15)

**First Choice: Loan Defaults**

1) Source of data

https://towardsdatascience.com/machine-learning-predicting-bank-loan-defaults-d48bffb9aee2

2) Brief description of data

The dataset documents loan holders' information while recording whether the loan is bad or not. 

3) What is the target?

I will be targeting the Bad Loan column for my predictions to determine if the loan will be in good standing or is probable to default.

4) Is this a classification or regression problem?

The dataset is a classification problem.

5) How many features?

There will be 13 features: Grade, Annual Income, Short Employment (1 year or less), Employment Length Number (in years), Home Ownership, Debt-to-Income Ratio, Purpose, Term, Last Delinquency, Last Major Derog (bad rating within 90 days), Revolving Line (percent credit utilization), Total Received Late Fee, and Overdraft Ratio.

6) How many rows of data.

There are 20,000 rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

I believe there will be minimal error or challenges cleaning, exploring and modeling this dataset.

In [19]:
path = "/content/train.csv.zip"
vld = pd.read_csv(path)
vld.head()

UniqueID  disbursed_amount  asset_cost    ltv  branch_id  supplier_id  \
0    420825             50578       58400  89.55         67        22807   
1    537409             47145       65550  73.23         67        22807   
2    417566             53278       61360  89.63         67        22807   
3    624493             57513       66113  88.48         67        22807   
4    539055             52378       60300  88.39         67        22807   

   manufacturer_id  Current_pincode_ID Date.of.Birth Employment.Type  ...  \
0               45                1441      01-01-84        Salaried  ...   
1               45                1502      31-07-85   Self employed  ...   
2               45                1497      24-08-85   Self employed  ...   
3               45                1501      30-12-93   Self employed  ...   
4               45                1495      09-12-77   Self employed  ...   

  SEC.SANCTIONED.AMOUNT  SEC.DISBURSED.AMOUNT  PRIMARY.INSTAL.AMT  \
0                     0                     0                   0   
1                     0                     0                1991   
2                     0                     0                   0   
3                     0                     0                  31   
4                     0                     0                   0   

   SEC.INSTAL.AMT  NEW.ACCTS.IN.LAST.SIX.MONTHS  \
0               0                             0   
1               0                             0   
2               0                             0   
3               0                             0   
4               0                             0   

   DELINQUENT.ACCTS.IN.LAST.SIX.MONTHS  AVERAGE.ACCT.AGE  \
0                                    0         0yrs 0mon   
1                                    1        1yrs 11mon   
2                                    0         0yrs 0mon   
3                                    0         0yrs 8mon   
4                                    0         0yrs 0mon   

   CREDIT.HISTORY.LENGTH  NO.OF_INQUIRIES  loan_default  
0              0yrs 0mon                0             0  
1             1yrs 11mon                0             1  
2              0yrs 0mon                0             0  
3              1yrs 3mon                1             1  
4              0yrs 0mon                1             1  

[5 rows x 41 columns]

In [20]:
vld.shape

(233154, 41)

**Second Choice: Bitcoin Stock**

1) Source of data

https://www.kaggle.com/datasets/sneharshinde/ltfs-av-data?select=train.csv

2) Brief description of data

The dataset provided reflects Vehicle Loan Defaults through L&T Finhack. There are 41 columns of data that will be reduced to relavant information.

3) What is the target?

I will be targeting the loan default column to determine if the borrower will default or not.

4) Is this a classification or regression problem?

The dataset is a clasification problem.

5) How many features?

There will be 36 features: Disbursed amount, Asset cost, Loan-to-value,  branch_id, supplier_id, manufacturer_id, Current_pincode_ID, Employment_Type, DisbursalDate, MobileNo_Avl_Flag, Aadhar_flag, PAN_flag, VoterID_flag, Driving_flag, Passport_flag, PERFORM_CNS_SCORE, PERFORM_CNS_SCORE_DESCRIPTION, PRI_NO_OF_ACCTS, PRI_ACTIVE_ACCTS, PRI_OVERDUE_ACCTS, PRI_CURRENT_BALANCE, PRI_SANCTIONED_AMOUNT, PRI_DISBURSED_AMOUNT, SEC_NO_OF_ACCTS, SEC_ACTIVE_ACCTS, SEC_OVERDUE_ACCTS, SEC_CURRENT_BALANCE, SEC_SANCTIONED_AMOUNT, SEC_DISBURSED_AMOUNT, PRIMARY_INSTAL_AMT, SEC_INSTAL_AMT, NEW_ACCTS_IN_LAST_SIX_MONTHS, DELINQUENT_ACCTS_IN_LAST_SIX_MONTHS, AVERAGE_ACCT_AGE, CREDIT_HISTORY_LENGTH, NO_OF_INQUIRIES.

6) How many rows of data.

There are 233,154 rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

I can see the amount of data available could create an issue when modeling. Additionally, exploring the data may take a long time.